benchmarks/nel/scripts/datasets/utils.py

```diff
@@ -0,0 +1,188 @@
+""" Utilities for NEL benchmark. """
+
+import copy
+import urllib.parse
+from typing import Tuple, Set, List, Dict, Optional, Union
+
+import requests
+import tqdm
+from spacy.tokens import Token, Doc, Span
+
+# todo @RM replace with pydantic schemas
+ENTITIES_TYPE = Dict[str, Dict[str, Union[Set[str], str, int]]]
+ANNOTATIONS_TYPE = Dict[str, List[Dict[str, Union[Set[str], str, int]]]]
+
+# Max. allowed Wikipedia API batch size with descriptions.
+MAX_WIKI_API_BATCH_SIZE = 20
+
+
+def _does_token_overlap_with_annotation(token: Token, annot_start: int, annot_end: int) -> bool:
+    """ Checks whether token overlaps with annotation span.
+    token (Token): Token to check.
+    annot_start (int): Annotation's start index.
+    annot_end (int): Annotation's end index.
+    RETURNS (bool): Whether token overlaps with annotation span.
+    """
+
+    return annot_start <= token.idx <= annot_end or token.idx <= annot_start <= token.idx + len(token)
+
+
+def _resolve_wiki_titles(
+    entities: ENTITIES_TYPE,
+    entity_titles: Optional[Set[str]] = None,
+    batch_size: int = MAX_WIKI_API_BATCH_SIZE
+) -> Tuple[ENTITIES_TYPE, Set[str], Dict[str, str]]:
+    """
+    Resolves Wikipedia titles to Wikidata IDs. Also fetches descriptions for entities.
+
+    entities (ENTITIES_TYPE): Entities to resolve.
+    entity_titles (Set[str]): List of entity titles to resolve.
+    batch_size (int): Number of entity titles to resolve in the same API request. Between 1 and 50.
+    RETURNS (Dict[str, str]): Mapping of titles to entity IDs.
+    """
+
+    assert 1 <= batch_size <= MAX_WIKI_API_BATCH_SIZE, \
+        f"Batch size has to be between 1 and {MAX_WIKI_API_BATCH_SIZE}."
+    entity_titles = list(entity_titles if entity_titles else entities.keys())
+    pbar = tqdm.tqdm(total=len(entity_titles))
```

Comment:
The `tqdm` progress bar could use a title here so you know what it is exactly you're waiting for - something like "Processing relevant Wikidata entities"?

Reply:
Or alternatively just a log statement to go before it. Right now the output is:
```

================================= create_kb =================================
Running command: '...\python.exe' ./scripts/create_kb.py reddit en_core_web_md
100%|██████████| 5167/5167 [01:42<00:00, 50.60it/s]
Trying to salvage 12 failed lookups
100%|██████████| 12/12 [00:02<00:00,  4.65it/s]
Constructing knowledge base with 5022 entries
```
So there's currently no explanation about the first progress bar.

Reply:
And one final nitpick on that: it also finishes with the "Constructing ..." log statement. Maybe we want to add one more log statement noting it finished succesfully? Otherwise it might look like it crashed/stopped mid-through.